### PR TITLE
1278/satori over puppeteer

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

- Fixes https://github.com/Shopify/polaris-internal/issues/1278

Our existing `gen-og-images` script creates an og-image layout using HTML markup and then runs puppeteer to take snapshots of each generated layout. This is relatively slow, this PR aims to swap this logic out for [Satori](https://github.com/vercel/satori) vercel's library for doing the exact same markup to og-image procedure but without having to spin up a puppeteer instance. 

### WHAT is this pull request doing?
* Convert our HTML markup to JSX 
* Convert `<style>` tag CSS to inline css for each of our JSX elements. (This is a limitation of satori, in that it does not support global <style> tags
* Remove puppeteer logic in favor of a single call to satori 
```tsx
satori(someMarkup, someConfig)
```
* Leverage [@inter-ui](https://www.npmjs.com/package/inter-ui) to download non-variable versions of Inter for use in our generated layout. (A limitation of the underlying library that Satori leverages for type support (opentype.js) is that it does not support variable fonts) 
* Leverage [@napi-rs/image](https://www.npmjs.com/package/@napi-rs/image) to convert our generated svg from satori into a png file for use as an og-image. 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by gwyneplaine** [comment]
The resultant og-images are slightly different due to: 
* Lack of [CSS three dimensional transform](https://github.com/vercel/satori?tab=readme-ov-file#css) support in satori
* Reconstructed layout to account for long titles.

| Before | After |
|--------|--------|
| view-transitions| view-transitions |
| account-connection | account-connection |
| experience-values |experience-values|

We're not altogether happy with how the component og-image looks, happy for designers to weigh in here and give this a lil more love 👍. Provided the design can work with the [limited CSS support](https://github.com/vercel/satori?tab=readme-ov-file#css) in Satori.

File: polaris.shopify.com/content/components/index.mdx [hunk 1]
```diff
@@ -2,7 +2,6 @@
 title: Components
 order: 7
 icon: AppsMajor
-previewImg: /images/components.png
```
Comment (jesstelford) [inline polaris.shopify.com/content/components/index.mdx:5]: This file doesn't exist.

File: polaris.shopify.com/package.json [hunk 2]
```diff
@@ -84,15 +85,16 @@
     "generact": "^0.4.0",
     "get-site-urls": "3.0.0-alpha.1",
     "globby": "^11.1.0",
+    "inter-ui": "3.19.3",
```
Comment (jesstelford) [inline polaris.shopify.com/package.json:88]: The `satori` package used to generate open graph images doesn't support woff2 or variable width fonts (the one on Shopify's CDN is both of those), so instead we use the version pushed to npm which has a non-variable woff set for us to use.

File: polaris.shopify.com/tsconfig.json [hunk 3]
```diff
@@ -37,7 +37,8 @@
     "transpileOnly": true,
     "files": true,
     "compilerOptions": {
-      "module": "CommonJS"
+      "module": "CommonJS",
+      "jsx": "react"
```
Comment (jesstelford) [inline polaris.shopify.com/tsconfig.json:41]: Without this, the JSX required by `satori` will fail to be compiled.

**Review by jesstelford — APPROVED** [review]

**Comment by gwyneplaine** [comment]
Og-images now all use the same icon and layout, as per discussion with @tjonx and @heyjoethomas 
| Updated og-images |
|--------|
| account-connection |
| view-transitions |
| actionable-language |

**Review by alex-page — APPROVED** [review]

**Pull request merged**